### PR TITLE
SuggestedCacheKey to use client ID and tenant ID

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fix for #1948

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs [hunk 1]
```diff
@@ -16,16 +16,19 @@ public static string GetKeyFromRequest(AuthenticationRequestParameters requestPa
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenSilent)
             {
-                return requestParameters.Account?.HomeAccountId?.Identifier;
+                return requestParameters.Account?.HomeAccountId?.Identifier == null ?
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:19]: @jmprieur - if you could review just this file, it captures all the actual changes, the rest are tests ...

Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs:19]: Do we have the risk that the first time the pre-access event is called for OBO, we get a cache key, and the second time we get a different cache key? [we'd want it to be invariant and the same in all calls to before/after access]

**Review by jmprieur — COMMENTED** [review]
I'm not sure about the tests.
We want all the calls to both before access and after access to have the same cache key

I think that testing if the home account id is null we induce a different cache key in the first call for OBO (as we don't know the user yet) ?

**Comment by bgavrilMS** [comment]
Discussed offline with @jmprieur and decided to not take this for now as Microsoft.Identity.Web is in "ask mode"

**Pull request closed without merging**